src/commands/commands.h

```diff
@@ -0,0 +1,34 @@
+#pragma once
+
+#include <map> 
+#include <string>
+#include <vector>
+
+typedef int commandFunction(const std::vector<std::string>);
+
+struct CommandInfo {
+    const std::string name; 
+    const std::string usage;
+    const commandFunction* command;
+};
+
+
+class Commands {
+    private:
+        std::map<std::string, const CommandInfo*> cmdList;
+
+    public:
+        Commands();
+        ~Commands();
+        void add(const CommandInfo* command);
```

Comment:
Since the class `Commands` takes ownership of the `command`, we can pass i by unique pointer. 
That said, we don't need a pointer at all, we can just take the value in.